packages/core/strapi/src/cli/commands/cloud/deploy-project.ts

```diff
@@ -0,0 +1,147 @@
+import { createCommand } from 'commander';
+import fs from 'fs';
+import path from 'path';
+import type { CLIContext, StrapiCommand } from '../../types';
+import { runAction } from '../../utils/helpers';
+import { apiConfig } from './config/api';
+import { getStoragePath } from './config/local';
+import { action as createProjectAction } from './create-project';
+import { cloudApiFactory, ProjectInfos } from './services/cli-api';
+import * as localSave from './services/strapi-info-save';
+import { compressFilesToTar } from './utils/compress-files';
+import { notificationServiceFactory } from './utils/notification-service';
+import { tokenServiceFactory } from './utils/token';
+
+const FILE_SIZE_LIMIT = 100 * 1024 * 1024; // 100MB
+
+function getPackageJson({ logger }: CLIContext, folderPath: string): Record<string, any> | null {
+  const packageJsonPath = path.join(folderPath, 'package.json');
+
+  if (!fs.existsSync(packageJsonPath)) {
+    logger.error(
+      "This project seems to be missing a 'package.json' file. It's required for project configuration."
+    );
+    return null;
+  }
+
+  try {
+    const packageJson = require(packageJsonPath);
+
+    if (!packageJson.strapi || !packageJson.strapi.uuid) {
+      logger.error(
+        'This project appears to be missing a Strapi configuration. Please check your project setup and try again.'
+      );
+      return null;
+    }
+    return packageJson;
+  } catch (error: Error | unknown) {
+    logger.debug(error);
+    if (error instanceof Error && error.message.includes('Unexpected token')) {
+      logger.error(
+        'The package.json file seems to have invalid formatting. Please check for syntax errors and try again.'
+      );
+    } else {
+      logger.error('Unable to proceed: The package.json file does not contain valid JSON data.');
+    }
+  }
+  return null;
+}
+
+async function upload(ctx: CLIContext, project: ProjectInfos, token: string) {
```

Comment:
Would `deploy` be clearer than `upload`?

Reply:
The deploy is done by cloud platform-api. Here we only upload the compressed package to it. Does it sounds reasonable to you?
FYI @nathan-pichon

Reply:
I think it should remain "upload" 🙂 
Or maybe "uploadCompressedProject"
Later we will have "displayBuildLogs" I guess 🤔

Reply:
Sounds good to me :)